crates/factors/src/factor.rs

```diff
@@ -0,0 +1,163 @@
+use std::any::Any;
+
+use anyhow::Context;
+
+use crate::{
+    App, InstancePreparers, Linker, ModuleLinker, PrepareContext, RuntimeConfig, RuntimeFactors,
+};
+
+pub trait Factor: Any + Sized {
+    /// Per-app state for this factor.
+    ///
+    /// See [`Factor::configure_app`].
+    type AppState: Default;
+
+    /// The [`FactorInstancePreparer`] for this factor.
```

Comment:
Many of the doc comments on this trait need to be updated to remove reference to `FactorInstancePreparer`.

Reply:
...unless we bring it back :slightly_smiling_face: